beacon-chain/forkchoice/protoarray/optimistic_sync.go

```diff
@@ -88,3 +88,204 @@ func (f *ForkChoice) Optimistic(ctx context.Context, root [32]byte, slot types.S
 	f.store.nodesLock.RUnlock()
 	return f.Optimistic(ctx, root, slot)
 }
+
+// UpdateSyncedTips updates the synced_tips map when the block with the given root becomes VALID
+func (f *ForkChoice) UpdateSyncedTips(ctx context.Context, root [32]byte) error {
+	f.store.nodesLock.RLock()
+	defer f.store.nodesLock.RUnlock()
+	// We can only update if given root is in fork choice
+	index, ok := f.store.nodesIndices[root]
+	if !ok {
+		return errInvalidNodeIndex
+	}
+
+	// We can only update if root is a leaf in fork choice
+	node := f.store.nodes[index]
+	if node.bestChild != NonExistentNode {
+		return errInvalidBestChildIndex
+	}
+
+	// Stop early if the root is part of validated tips
+	f.syncedTips.Lock()
+	defer f.syncedTips.Unlock()
+	_, ok = f.syncedTips.validatedTips[root]
+	if ok {
+		return nil
+	}
+
+	// Cache root and slot to validated tips
+	f.syncedTips.validatedTips[root] = node.slot
```

Comment:
Shouldn't you do this _after_ the validation?

What if the context times out before validation has been completed?

Reply:
Oh sorry it took me a little to understand this comment is about a different PR. So this method takes a root of a block that we already know from the EL that it's VALID, since it's a tip of the fork choice tree then it has to be a synced tip no matter what